src/main/java/com/iexec/worker/feign/LoginService.java

```diff
@@ -48,48 +45,41 @@ public String getToken() {
         return jwtToken;
     }
 
-    @Override
     public String login() {
         expireToken();
 
         String workerAddress = credentialsService.getCredentials().getAddress();
         ECKeyPair ecKeyPair = credentialsService.getCredentials().getEcKeyPair();
 
-        String challenge = getLoginChallenge(workerAddress);
+        ResponseEntity<String> challengeResponse = coreClient.getChallenge(workerAddress);
+        if (!challengeResponse.getStatusCode().is2xxSuccessful()) {
+            log.error("Cannot login, failed to get challenge [status:{}]", challengeResponse.getStatusCode());
+            return "";
+        }
+        String challenge = challengeResponse.getBody();
         if (StringUtils.isEmpty(challenge)) {
-            log.error("Cannot login since challenge is empty [challenge:{}]", challenge);
+            log.error("Cannot login, challenge is empty [challenge:{}]", challenge);
             return "";
         }
 
         Signature signature = SignatureUtils.hashAndSign(challenge, workerAddress, ecKeyPair);
-        String token = requestLogin(workerAddress, signature);
+        ResponseEntity<String> tokenResponse = coreClient.login(workerAddress, signature);
+        if (!tokenResponse.getStatusCode().is2xxSuccessful()) {
+            log.error("Cannot login, failed to get token [status:{}]", tokenResponse.getStatusCode());
+            return "";
+        }
+        String token = tokenResponse.getBody();
         if (StringUtils.isEmpty(token)) {
-            log.error("Cannot login since token is empty [token:{}]", token);
+            log.error("Cannot login, token is empty [token:{}]", token);
             return "";
         }
 
+        log.info("Retrieved new token {}", token);
```

Comment:
Not sure about this line. Isn't it too risky to log a token? An attacker may be able to retrieve logs and use a token to do malicious things.